src/zcash/Address.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef _ZCADDRESS_H_
+#define _ZCADDRESS_H_
+
+#include "uint256.h"
+#include "serialize.h"
+
+namespace libzcash {
+
+class PaymentAddress {
+public:
+    uint256 a_pk;
+    uint256 pk_enc;
+
+    PaymentAddress() : a_pk(), pk_enc() { }
+    PaymentAddress(uint256 a_pk, uint256 pk_enc) : a_pk(a_pk), pk_enc(pk_enc) { }
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action, int nType, int nVersion) {
+        unsigned char leadingByte = 0x92;
+        READWRITE(leadingByte);
+
+        if (leadingByte != 0x92) {
+            throw std::ios_base::failure("unrecognized payment address version");
```

Comment:
"unrecognized payment address lead byte" (it may not be a version as such; there may be different kinds of key with different features).